src/Uno.UI/UI/Xaml/UIElement.Skia.cs

```diff
@@ -0,0 +1,300 @@
+
+			while (current != null && current != root)
+			{
+				current = this.GetParent();
+			}
+
+			return current == root;
+		}
+
+		private void OnAddingChild(UIElement child)
+		{
+			if (IsLoaded)
+			{
+				child.OnElementLoaded();
+			}
+		}
+
+		partial void OnOpacityChanged(DependencyPropertyChangedEventArgs args)
+		{
+			UpdateOpacity();
+		}
+
+		partial void OnIsHitTestVisibleChangedPartial(bool oldValue, bool newValue)
+		{
+			UpdateHitTest();
+		}
+
+		private void UpdateOpacity() => Visual.Opacity = Visibility == Visibility.Visible ? (float)Opacity : 0;
+
+		private void UpdateChildVisual(UIElement child)
+		{
+			Visual.Children.InsertAtTop(child.Visual);
+		}
+
+		internal UIElement RemoveChild(UIElement child)
+		{
+			_children.Remove(child);
+			child.SetParent(null);
+
+			if (Visual != null)
+			{
+				Visual.Children.Remove(child.Visual);
+			}
+
+			return child;
+		}
+
+		internal void ClearChildren()
+		{
+			foreach (var child in _children)
+			{
+				child.SetParent(null);
+				// OnChildRemoved(child);
+			}
+
+			_children.Clear();
+			InvalidateMeasure();
+		}
+
+		internal UIElement FindFirstChild() => _children.FirstOrDefault();
+
+		public string Name { get; set; }
+
+		partial void InitializeCapture();
+
+		internal bool IsPointerCaptured { get; set; }
+
+		internal virtual bool IsEnabledOverride() => true;
+
+		public virtual IEnumerable<UIElement> GetChildren() => _children;
+
+		public IntPtr Handle { get; set; }
+
+		internal Windows.Foundation.Point GetPosition(Point position, global::Windows.UI.Xaml.UIElement relativeTo)
+		{
+			throw new NotSupportedException();
+		}
+
+		protected virtual void OnVisibilityChanged(Visibility oldValue, Visibility newVisibility)
+		{
+			UpdateOpacity();
+
+			if (newVisibility == Visibility.Collapsed)
+			{
+				_desiredSize = new Size(0, 0);
+				_size = new Size(0, 0);
+			}
+		}
+
+		partial void OnRenderTransformSet()
+		{
+		}
+
+		internal void ArrangeVisual(Rect finalRect, bool clipToBounds, Rect? clippedFrame = default)
+		{
+			LayoutSlotWithMarginsAndAlignments =
+				VisualTreeHelper.GetParent(this) is UIElement parent
+					? finalRect.DeflateBy(parent.GetBorderThickness())
+					: finalRect;
+
+			var oldFinalRect = _currentFinalRect;
+			_currentFinalRect = finalRect;
+
+			var oldRect = oldFinalRect;
+			var newRect = finalRect;
+			if (oldRect != newRect)
+			{
+				if (
+					newRect.Width < 0
+					|| newRect.Height < 0
+					|| double.IsNaN(newRect.Width)
+					|| double.IsNaN(newRect.Height)
+					|| double.IsNaN(newRect.X)
+					|| double.IsNaN(newRect.Y)
+				)
+				{
+					throw new InvalidOperationException($"{this}: Invalid frame size {newRect}. No dimension should be NaN or negative value.");
+				}
+
+				Rect? getClip()
+				{
+					// Disable clipping for Scrollviewer (edge seems to disable scrolling if 
+					// the clipping is enabled to the size of the scrollviewer, even if overflow-y is auto)
+					if (this is Controls.ScrollViewer)
```

Comment:
Codacy found an issue: [Offload the code that's conditional on this 'is' test to the appropriate subclass and remove the test.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5872302)